src/NServiceBus.Transport.SQS/IAmazonSqsMessageTranslationStrategy.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace NServiceBus.Transport.SQS
+{
+    using System.Collections.Generic;
+    using Amazon.SQS.Model;
+    using static TransportHeaders;
+
+    /// <summary>
+    /// Configures how the incoming SQS transport message is extracted
+    /// </summary>
+    public interface IAmazonSqsIncomingMessageExtractor
+    {
+        /// <summary>
+        /// Performs the SQS transport message extraction
+        /// </summary>
+        /// <param name="receivedMessage"></param>
+        /// <param name="messageId"></param>
+        /// <param name="headers"></param>
+        /// <param name="s3BodyKey"></param>
+        /// <param name="body"></param>
+        /// <returns></returns>
+        bool TryExtractMessage(Message receivedMessage, string messageId, out Dictionary<string, string> headers, out string s3BodyKey, out string body);
+    }
+
+    class DefaultAmazonSqsIncomingMessageExtractor : IAmazonSqsIncomingMessageExtractor
```

Comment:
What about making the `DefaultAmazonSqsIncomingMessageExtractor` public?
It would simplify users' life, they could handle only their special cases, and fallback to the default extractor if needed.

Reply:
In theory a customer that wants to do this could already do so like this:

```csharp
sqs.IncomingMessageExtractor = new MyCustomIncomingMessageExtractor(sqs.IncomingMessageExtractor);
```

Is it worth adding this class to the public interface?

Reply:
Ah, good point. I missed that. No need to make it public.